### PR TITLE
fix(rdb): populate waitinstance on related resources

### PR DESCRIPTION
**Pull request by Monitob**
- fix(rdb): populate waiting and retry
- fix(rdb): get region from id and retry on database creation

### CODE REVIEW

File: scaleway/resource_rdb_user.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func resourceScalewayRdbUserCreate(ctx context.Context, d *schema.ResourceData,
 
 	ins, err := waitInstance(ctx, rdbAPI, region, instanceID)
 	if err != nil {
-		return diag.FromErr(err)
+		diag.FromErr(err)
```
Comment (remyleone) [inline scaleway/resource_rdb_user.go:73]: Why is this change needed?

File: scaleway/resource_rdb_instance.go [hunk 2]
```diff
@@ -321,7 +321,7 @@ func resourceScalewayRdbInstanceCreate(ctx context.Context, d *schema.ResourceDa
 		}
 
 		_, err = waitInstance(ctx, rdbAPI, region, res.ID)
-		if err != nil && !is404Error(err) {
+		if err != nil {
```
Comment (remyleone) [inline scaleway/resource_rdb_instance.go:324]: Why are those changes needed? Should this kind of change be present on all other resources that check for 404 errors?

File: scaleway/resource_rdb_database.go [hunk 3]
```diff
@@ -58,52 +60,91 @@ func resourceScalewayRdbDatabase() *schema.Resource {
...
 		return diag.FromErr(err)
 	}
 
-	d.SetId(resourceScalewayRdbDatabaseID(region, expandID(instanceID), res.Name))
+	d.SetId(resourceScalewayRdbDatabaseID(region, instanceID, db.Name))
 
 	return resourceScalewayRdbDatabaseRead(ctx, d, meta)
 }
 
+func getDatabase(ctx context.Context, api *rdb.API,
```
Comment (remyleone) [inline scaleway/resource_rdb_database.go:108]: Could the signature be on a single line to be diff friendly?

File: scaleway/resource_rdb_database.go [hunk 4]
```diff
@@ -58,52 +60,91 @@ func resourceScalewayRdbDatabase() *schema.Resource {
...
 	createReq := &rdb.CreateDatabaseRequest{
 		Region:     region,
 		InstanceID: instanceID,
 		Name:       d.Get("name").(string),
 	}
 
-	res, err := rdbAPI.CreateDatabase(createReq, scw.WithContext(ctx))
+	var db *rdb.Database
+	//  wrapper around StateChangeConf that will just retry the database creation
+	err = resource.RetryContext(context.Background(), 2*time.Minute, func() *resource.RetryError {
```
Comment (remyleone) [inline scaleway/resource_rdb_database.go:81]: Could this 2*time.Minute be configurable?

File: scaleway/resource_rdb_user.go (hunk 1 above)
Comment (Monitob) [reply scaleway/resource_rdb_user.go]: oops copy/paste error. I've juste added the return

File: scaleway/resource_rdb_instance.go (hunk 2 above)
Comment (Monitob) [reply scaleway/resource_rdb_instance.go:324]: After a review of code, I remove is404Error where I consider where I think is necessary that the instance must exist. Which protect other request with an InstanceID not null and in a stable mode.

**Review by remyleone — APPROVED** [review]
